### DISCUSSION TITLE
Add source generator from local dll into C# project?

### DISCUSSION THREAD

**Discussion by acaly** [dotnet/roslyn#49820] [Q&A] [answered]
Hi! I am not very familiar with Roslyn analyzer before, but recently have tried a lot on source generator. It is brilliant! From the tutorial it seems source generator can be added as a project reference or nuget package. I wonder is there an easier way, to simply add a local dll file? I think it would be useful to include the dll directly with source code in some cases, if it is shared by multiple projects but itself does not change very often (and of course, I don't want to publish a nuget package). Thanks!

**Comment by ufcpp** [comment]
https://github.com/dotnet/roslyn-sdk/commit/52f4c4e75619f4a161cd8418bcd30d7b7d3ee504#diff-25d4a52c171c6bbd87583d2ade5d4c118a243eb59c1361a1c75c9a4d5c1f56df ?

**Comment by PathogenDavid** [comment, marked as answer]
You can add an analyzer/source generator directly by adding an `Analyzer` item. IE:

```xml
<ItemGroup>
  <Analyzer Include="Path/To/Your/Analyzer.dll" />
</ItemGroup>
```

**Reply by acaly** [reply]
Cool! Thank you!

**Reply by SSASA-MOVIL** [reply]
Perfect!
